File: java-example/src/main/java/org/luovuutesi/moduleexample/java/ExampleModule.java
package org.luovuutesi.moduleexample.java;

import org.jetbrains.annotations.NotNull;
import org.luovuutesi.api.Module;

public class ExampleModule implements Module {
    @NotNull
    @Override
    public String getModuleId() {
        return "example-module-java";
    }

    @Override
    public void start() {
        System.out.println("Started Java Example Module!");
    }

    @Override
    public void stop() {
        System.out.println("Stopped Java Example Module!");
    }
}

File: kotlin-example/src/main/kotlin/org/luovuutesi/moduleexample/kotlin/ExampleModule.kt
package org.luovuutesi.moduleexample.kotlin

import org.luovuutesi.api.Module

class ExampleModule : Module {
    override val moduleId = "example-module-kotlin"

    override fun start() {
        println("Started Kotlin Example Module!")
    }

    override fun stop() {
        println("Stopped Kotlin Example Module!")
    }
}